src/util/hostname.cpp

```diff
@@ -0,0 +1,30 @@
+#include <string>
+
+#include "hostname.hpp"
+
+#ifdef __linux__
+extern "C" {
+    #include <unistd.h>
+}
+#endif
+
+namespace nest {
+namespace mc {
+namespace util {
+
+#ifdef __linux__
+std::string hostname() {
+    char name[128];
+    auto result = gethostname(name, sizeof(name));
+    return result? "unknown": name;
+}
+#else
+std::string hostname() {
+    return "unknown";
```

Comment:
I still think we need a better signal for failure to determine hostname, e.g. empty string or a string with invalid hostname characters in it (e.g. "<unknown>".)